jwst/datamodels/tests/test_open.py

```diff
@@ -257,3 +257,12 @@ def t_path(partial_path):
     """Construction the full path for test files"""
     test_dir = os.path.join(os.path.dirname(__file__), 'data')
     return os.path.join(test_dir, partial_path)
+
+
+def test_open_asdf_datamodel_class(tmpdir):
+    path = str(tmpdir.join("image_model.asdf"))
```

Comment:
Could you look at parametrizing this test so that it also gets checked for `fits` files?  So something like:

```
@pytest.mark.parametrize("suffix", ["asdf", "fits"])
def test_open_asdf_datamodel_class(tmpdir, suffix):
      path = str(tmpdir.join(f"image_model.{suffix}"))
```

And also check for the ".asdf" case that when there is no `model_type` set, that you get a `DataModel` returned.  Maybe there's already an existing test that is checking the ".fits" case?  Feel free to add a new test that checks both.  Or do it in the same test above as a separate check.

Looking at codecov, those areas of `_class_from_model_type` are not covered.

https://codecov.io/gh/spacetelescope/jwst/compare/a8d5086eb22efc3a19ace4c1dadce63db89f9af2...f4b59215e765618cab1ca4a96dfec4771a20caed/diff

Reply:
Added parametrization for `test_open_asdf_datamodel_class` and added tests `test_open_asdf_no_datamodel_class` and `test_open_asdf` to increase code coverage.